src/controller/robot/controller.go

```diff
@@ -273,24 +275,19 @@ func (d *controller) populatePermissions(ctx context.Context, r *Robot) error {
 	return nil
 }
 
-func (d *controller) setProjectID(ctx context.Context, r *Robot) error {
+// set the project info if it's a project level robot
+func (d *controller) setProject(ctx context.Context, r *Robot) error {
 	if r == nil {
 		return nil
 	}
-	var projectID int64
-	switch r.Level {
-	case LEVELSYSTEM:
-		projectID = 0
-	case LEVELPROJECT:
+	if r.Level == LEVELPROJECT {
```

Comment:
When the r.Level is wrong, there is no error. and the robot account can be created.  Is it an expected behaviour?

Reply:
For the input checking, all of them are done by API handler. And this method is just to set the project information, if it's a system level, the project ID will be 0, it's a expected behavior.